xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -473,6 +477,31 @@ bool CProcessorHD::OpenProcessor()
   color.YCbCr = { 0.0625f, 0.5f, 0.5f, 1.0f }; // black color
   m_pVideoContext->VideoProcessorSetOutputBackgroundColor(m_pVideoProcessor.Get(), TRUE, &color);
 
+  if (m_hasMetadataHDR10Support)
```

Comment:
Please add more conditions here, similar to:

```c++
 DXGI_ADAPTER_DESC ad{};
 DX::DeviceResources::Get()->GetAdapterDesc(&ad);
bool streamIsHDR = (picture.color_primaries == AVCOL_PRI_BT2020) &&
                           (picture.color_transfer == AVCOL_TRC_SMPTE2084 ||
                            picture.color_transfer == AVCOL_TRC_ARIB_STD_B67);

  if (m_hasMetadataHDR10Support && ad.VendorId == PCIV_AMD && streamIsHDR)
```

This could cause side effects if is enabled on all systems.

Reply:
I understand